core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java

```diff
@@ -0,0 +1,424 @@
+        }
+      };
+
+  /** Handle to the block master. */
+  private final BlockMaster mBlockMaster;
+
+  /** The clock to use for determining the time. */
+  private final Clock mClock = new SystemClock();
+
+  /** The master configuration store. */
+  private final ServerConfigurationStore mMasterConfigStore = new ServerConfigurationStore();
+  /** The worker configuration store. */
+  private final ServerConfigurationStore mWorkerConfigStore = new ServerConfigurationStore();
+  /** The server-side configuration checker. */
+  private final ServerConfigurationChecker mConfigChecker =
+      new ServerConfigurationChecker(mMasterConfigStore, mWorkerConfigStore);
+
+  /** Keeps track of standby masters which are in communication with the leader master. */
+  private final IndexedSet<MasterInfo> mMasters =
+      new IndexedSet<>(ID_INDEX, ADDRESS_INDEX);
+  /** Keeps track of standby masters which are no longer in communication with the leader master. */
+  private final IndexedSet<MasterInfo> mLostMasters =
+      new IndexedSet<>(ID_INDEX, ADDRESS_INDEX);
+
+  /** The address of this master. */
+  private final Address mMasterAddress = new Address()
+      .setHost(Configuration.get(PropertyKey.MASTER_HOSTNAME))
+      .setRpcPort(Configuration.getInt(PropertyKey.MASTER_RPC_PORT));
+
+  /** The connect address for the rpc server. */
+  private final InetSocketAddress mRpcConnectAddress
+      = NetworkAddressUtils.getConnectAddress(NetworkAddressUtils.ServiceType.MASTER_RPC);
+
+  /** The manager of safe mode state. */
+  private final SafeModeManager mSafeModeManager;
+
+  /** The start time for when the master started serving the RPC server. */
+  private final long mStartTimeMs;
+
+  /** The master ID for this master. */
+  private AtomicReference<Long> mMasterId = new AtomicReference<>(-1L);
+
+  /**
+   * Creates a new instance of {@link DefaultMetaMaster}.
+   *
+   * @param blockMaster a block master handle
+   * @param masterContext the context for Alluxio master
+   */
+  DefaultMetaMaster(BlockMaster blockMaster, MasterContext masterContext) {
+    this(blockMaster, masterContext, ExecutorServiceFactories
+        .fixedThreadPoolExecutorServiceFactory(Constants.META_MASTER_NAME, 2));
+  }
+
+  /**
+   * Creates a new instance of {@link DefaultMetaMaster}.
+   *
+   * @param blockMaster a block master handle
+   * @param masterContext the context for Alluxio master
+   * @param executorServiceFactory a factory for creating the executor service to use for running
+   *        maintenance threads
+   */
+  DefaultMetaMaster(BlockMaster blockMaster, MasterContext masterContext,
+      ExecutorServiceFactory executorServiceFactory) {
+    super(masterContext, new SystemClock(), executorServiceFactory);
+    mSafeModeManager = masterContext.getmSafeModeManager();
+    mStartTimeMs = masterContext.getStartTimeMs();
+    mBlockMaster = blockMaster;
+    mBlockMaster.registerLostWorkerFoundListener(this::lostWorkerFoundHandler);
```

Comment:
nit: could pass `mWorkerConfigStore::lostNodeFound` instead, similar for other registers on the next two lines